logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java

```diff
@@ -0,0 +1,656 @@
+import static com.linecorp.armeria.common.logback.BuiltInProperty.REMOTE_PORT;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.REQ_AUTHORITY;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.REQ_CONTENT_LENGTH;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.REQ_DIRECTION;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.REQ_METHOD;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.REQ_PATH;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.REQ_RPC_METHOD;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.REQ_RPC_PARAMS;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.RES_CONTENT_LENGTH;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.RES_RPC_RESULT;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.RES_STATUS_CODE;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.SCHEME;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.SSL_CIPHER_SUITE;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.SSL_PROTOCOL;
+import static com.linecorp.armeria.common.logback.BuiltInProperty.SSL_SESSION_ID;
+import static java.util.Objects.requireNonNull;
+
+import java.net.InetSocketAddress;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLSession;
+
+import org.slf4j.MDC;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.BaseEncoding;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.RpcResponse;
+import com.linecorp.armeria.common.http.HttpHeaderNames;
+import com.linecorp.armeria.common.http.HttpHeaders;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.ResponseLog;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import ch.qos.logback.core.spi.AppenderAttachable;
+import ch.qos.logback.core.spi.AppenderAttachableImpl;
+import io.netty.channel.Channel;
+import io.netty.util.AsciiString;
+import io.netty.util.AttributeKey;
+
+public class RequestContextExportingAppender extends UnsynchronizedAppenderBase<ILoggingEvent>
```

Comment:
Since this is Unsynchronized, do we need to make this class threadsafe (e.g. ConcurrentHashMap, computeIfAbsent)?